pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java

```diff
@@ -0,0 +1,428 @@
+            configLog(options);
+            TransactionDBOptions transactionDBOptions = new TransactionDBOptions();
+            db = TransactionDB.open(options, transactionDBOptions, dataPath.toString());
+
+            sequentialIdGenerator = new AtomicLong(0);
+            byte[] value = db.get(SEQUENTIAL_ID_KEY);
+            if (value != null) {
+                sequentialIdGenerator.set(toLong(value));
+            } else {
+                db.put(INSTANCE_ID_KEY, toBytes(sequentialIdGenerator.get()));
+            }
+
+            value = db.get(INSTANCE_ID_KEY);
+            if (value != null) {
+                instanceId = toLong(value) + 1;
+            } else {
+                instanceId = 0;
+            }
+            db.put(INSTANCE_ID_KEY, toBytes(instanceId));
+
+
+        } catch (RocksDBException e) {
+            throw new MetadataStoreException("Error open RocksDB database", e);
+        } catch (MetadataStoreException e) {
+            throw e;
+        } catch (Throwable t) {
+            throw new MetadataStoreException(t);
+        }
+
+        optionSync.setSync(true);
+        optionDontSync.setSync(false);
+
+        optionCache.setFillCache(true);
+        optionDontCache.setFillCache(false);
+    }
+
+    private void configLog(Options options) throws IOException {
+        // Configure file path
+        String logPath = System.getProperty("pulsar.log.dir", "");
+        Path logPathSetting;
+        if (!logPath.isEmpty()) {
+            logPathSetting = FileSystems.getDefault().getPath(logPath + "/rocksdb-log");
+            Files.createDirectories(logPathSetting);
+            options.setDbLogDir(logPathSetting.toString());
+        }
+
+        // Configure log level
+        String logLevel = System.getProperty("pulsar.log.level", "info");
+        switch (logLevel) {
+            case "debug":
+                options.setInfoLogLevel(InfoLogLevel.DEBUG_LEVEL);
+                break;
+            case "info":
+                options.setInfoLogLevel(InfoLogLevel.INFO_LEVEL);
+                break;
+            case "warn":
+                options.setInfoLogLevel(InfoLogLevel.WARN_LEVEL);
+                break;
+            case "error":
+                options.setInfoLogLevel(InfoLogLevel.ERROR_LEVEL);
+                break;
+            default:
+                log.warn("Unrecognized RockDB log level: {}", logLevel);
+        }
+
+        // Keep log files for 1month
+        options.setKeepLogFileNum(30);
+        options.setLogFileTimeToRoll(TimeUnit.DAYS.toSeconds(1));
+    }
+
+    @Override
+    public void close() throws Exception {
+        db.close();
+        optionSync.close();
+        optionDontSync.close();
+        optionCache.close();
+        optionDontCache.close();
+        emptyBatch.close();
+        super.close();
+    }
+
+    @Override
+    public CompletableFuture<Optional<GetResult>> get(String path) {
+        if (path.isEmpty()) {
```

Comment:
Maybe these checks could be done in the `AbstractMetadataStore`

Reply:
1. Added 'storeGet' in AbstractMetadataStore.
2. Moved 'put' to AbstractMetadataStore, as it's the same in all implementations.
3. Now, each metadata store implements just implement the 'storeXXX' version methods.